src/umap/umap.cpp

```diff
@@ -457,11 +508,16 @@ void __attribute ((constructor)) init_umap_lib( void )
 
   LOGGING_INIT;
 
+  total_mem_kb = get_mem_total();
+
   if ((page_size = sysconf(_SC_PAGESIZE)) == -1) {
     perror("ERROR: sysconf(_SC_PAGESIZE)");
     throw -1;
   }
 
+  // Default buffer if 80% of pages in memory
+  umap_buffer_size = ((total_mem_kb / (page_size / 1024)) * 80) / 100;
```

Comment:
Code repeated from line 379. Maybe make an inline function so if we change to 75% it is only changed in one place?